static/src/javascripts-legacy/projects/commercial/modules/article-aside-adverts.js

```diff
@@ -35,38 +35,30 @@ define([
         }
 
         $mainCol = $('.js-content-main-column');
+        $adSlot = $('.js-ad-slot', $col);
 
-        fastdom.read(function () {
+        if (!$adSlot.length) {
+            stop();
+            return Promise.resolve(false);
+        }
+
+        return fastdom.read(function () {
             return $mainCol.dim().height;
         })
         .then(function (mainColHeight) {
-            var adSlot, adType;
-            var $adSlotContainer = $('.js-ad-slot-container', $col[0]);
 
-            if (config.page.isImmersive) {
-                adType = mainColHeight >= minImmersiveArticleHeight ?
-                        'right' :
-                        'right-small';
-            } else {
-                adType = mainColHeight >= minArticleHeight ?
-                    'right-sticky' :
-                    'right-small';
+            // Should switch to 'right-small' MPU for short articles
+            if (mainColHeight < minContentHeight()) {
+              return fastdom.write(function () {
```

Comment:
You will need to return `$adSlot[0]` inside the function so it can be passed on to the following callback.